frontend/src/pages/Projects/Project/ProjectDetails/ProjectRegistrationSection/index.tsx

```diff
@@ -38,46 +42,132 @@ export const ProjectRegistrationSection = ({
   }, [query]);
 
   let content = <></>;
-  switch (editMode) {
-    case false:
-    default:
-      content = (
-        <>
-          <ReadForm fields={readFields(query)} />
-          {userHasEditCapability && (
-            <Box m={1} display="flex" justifyContent="flex-end" alignItems="flex-end">
-              <Button variant="contained" onClick={() => setEditMode(true)}>
-                Change Registration
-              </Button>
-            </Box>
-          )}
-        </>
-      );
+  switch (query?.data?.dbRowLock?.locked) {
```

Comment:
The switch statements structure is kind of confusing. If they're only checking for true/false you could just use if/elses and I think it would be a little easier to follow.

Reply:
yes in future tickets